scrapy/core/downloader/__init__.py

```diff
@@ -198,4 +204,5 @@ def _slot_gc(self, age=60):
         mintime = time() - age
         for key, slot in list(self.slots.items()):
             if not slot.active and slot.lastseen + slot.delay < mintime:
-                self.slots.pop(key).close()
+                inactive_slot = self.slots.pop(key)
+                inactive_slot.close()
```

Comment:
According to Codecov this part is not covered by tests (it probably was not covered by tests before.

It seems functionally equivalent to the code before the change, so maybe you could just revert this change.

I am OK with the change though, it is more readable. But if you are going to change it, it would be great if you could also extend test coverage for it. Although that may not be trivial.

Reply:
Yes. Sure (reverted changes). I previously made this change(assigning inactive slot to variable) for logging which is removed as result of previous feedback.